packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,180 @@
+import clsx from 'clsx'
+import {Text} from '../Text'
+import {Controls} from './components'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/video-player/base.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './VideoPlayer.module.css'
+
+type VideoPlayerProps = {
+  poster?: string
+  title: string
+  branding?: boolean
+  children: React.ReactElement | React.ReactElement[]
+  ref?: React.RefObject<HTMLVideoElement>
+} & React.HTMLProps<HTMLVideoElement>
+
+function Root({
+  poster,
+  title,
+  branding = true,
+  children,
+  className,
+  onPlay,
+  onPause,
+  onLoadedMetadata,
+  onPlaying,
+  ref,
+  ...rest
+}: VideoPlayerProps) {
+  const videoPlayerRef = useRef<HTMLVideoElement>(null)
+  const videoWrapperRef = useRef<HTMLDivElement>(null)
+  const videoRef = ref ? ref : videoPlayerRef
+  const [playing, setPlaying] = useState(false)
+  const [playedTime, setPlayedTime] = useState(0)
+  const [totalTime, setTotalTime] = useState(0)
+  const [trackInformation, setTrackInformation] = useState<TextTrackCueList | undefined>(undefined)
+
+  /* > Play / Pause                                                    */
+  const handleVideoPlayback = () => {
+    if (videoRef.current) {
+      if (videoRef.current.paused) {
+        videoRef.current.play()
+      } else {
+        videoRef.current.pause()
+      }
+    }
+  }
+
+  /* > Hide Default Captions                                                   */
+  useEffect(() => {
+    if (videoRef.current) {
+      for (let i = 0; i < videoRef.current.textTracks.length; i++) {
+        videoRef.current.textTracks[0].mode = 'hidden'
+      }
+    }
+  }, [videoRef])
+
+  /* > Set breakpoint classes                                                   */
+
+  useEffect(() => {
+    // TODO: This needs to run on load on play this doesn't get re-ran
+    const handleResize = () => {
+      const breakpoint = videoWrapperRef.current?.getBoundingClientRect().width
+      if (breakpoint && breakpoint < 650) {
+        videoWrapperRef.current.classList.add(styles.VideoPlayer__breakpointSmall)
+      } else {
+        videoWrapperRef.current?.classList.remove(styles.VideoPlayer__breakpointSmall)
+      }
+    }
+
+    const resizeObserver = new ResizeObserver(_ => {
+      if (videoWrapperRef.current) {
+        handleResize()
+      }
+    })
+
+    handleResize()
+
+    const currentRef = videoWrapperRef.current
+
+    resizeObserver.observe(videoWrapperRef.current as Element)
+    return () => {
+      resizeObserver.unobserve(currentRef as Element)
+    }
+  }, [videoWrapperRef, playing])
+
+  return (
+    <div
+      className={clsx(
+        styles.VideoPlayer__container,
+        styles.VideoPlayer__overlays,
+        !playing && styles.VideoPlayer__showOverlays,
+      )}
+      ref={videoWrapperRef}
+    >
+      {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
```

Comment:
This is needed because we can't inspect the children but the idea is that it is required and will be added on implementation.